Add logging utility for agent

flourishingtune opened the pull request:
The commit adds a logging utility to Zambeze agent as requested in #127. This allows using `head/tail` modes with `numlines`.

For example:
```
zambeze logs
zambeze logs --mode=head
zambeze logs --mode=tail --numlines=20
```

Defaults to `tail` mode with `10` lines. Doesn't support negative integers, which results in no lines logged.

This is tested with a 1G text file and the utility attempts to efficiently `head/tail` logs.

wigging commented:
The docstring for the `logs(mode, num_lines)` function is incomplete. It has two input parameters which are not defined in the docstring. Use the NumPy docstring style to define the parameters.

flourishingtune commented:
I have added docstring for the log method. As I was trying to debug a few other things, I thought it would be useful to have a `follow` feature on the `tail`. 

I added the `follow` segment on the update as well and it has been tested.

wigging commented:
May I ask why you want to contribute to this project? Are you using zambeze for work or for a personal project? You recently joined GitHub, you don't have any other repositories on here, and your profile doesn't say anything about yourself. This just seems odd to me. Your contributions are welcome, but I'm curious as to why you would want to contribute to this particular project.

wigging commented:
It would be helpful to add some documentation on how to use these features. Can you add a documentation page at `docs/source/get-started/logging.rst` that describes these logging commands? You can use the comments you posted in this pull request for the documentation.

wigging reviewed (approved):
I tested these logger features while running the ImageMagick example and everything works fine.

wigging commented:
@flourishingtune I'm not sure why the pytest and ruff checks are waiting for status reports. I updated the GitHub Actions workflows on the main branch to run on push and pull request events. Can you update your fork/branch and see if this enables the checks to run?

sankalpatimilsina12 commented:
@wigging 
I setup this profile mostly out of personal interests and wanting to contribute to open projects in my leisure time. I have been developing an orchestrated system myself with my other personal/work account and thought it would be nice to get my hands dirty on similar projects. Mostly just utilizing my leisure time.

I have updated the branch with the documentation page for logging. Seems like the checks are running now.

Sorry I may be a bit late to respond/work on things :)

wigging commented:
@sankalpatimilsina12 Thank you for the response. I suggest you use your **sankalpatimilsina12** profile for making contributions so they show up on your main profile. The **flourishingtune** account looks suspicious since it has no commit history and nothing on the profile page. Anyway, thank you for the contribution. I'll merge it with the main branch.
